src/thirdParty/registerPlugin.js

```diff
@@ -1,72 +1,35 @@
-import registerMixin from './registerMixin.js';
+import registerItems from './registerItems.js';
 import registerModule from './registerModule.js';
-import registerTool from './registerTool.js';
-import { pluginList } from './index.js';
+import { lib } from '../lib.js';
 
 /**
  * Register a plugin.
  *
- * @
- * @param  {Object} newMixin The registerPlugin to register.
+ * @param  {Object} plugin The plugin to register.
+ * @param  {boolean} [overwrite] Whether to overwrite, should a plugin of the
+ *                               same name already be registered.
  */
-export default function (plugin) {
+export default function (plugin, overwrite = false) {
   if (!plugin.name) {
     throw new Error (`plugin must have a unique name!`);
   }
 
-  if (isPluginNameRegistered(plugin.name)) {
-    console.warning(`A plugin with the name ${name} is already registered`);
+  if (!overwrite && isPluginNameRegistered(plugin.name)) {
+    console.warn(`Plugin ${plugin.name} already registered.`);
 
     return;
   }
 
-  // TODO: This isn't very atomic.
-  // What if a plugin clashes with all names, yet still gets registered?
-
-  if (plugin.mixins) {
-
-
-    const mixins = plugin.mixins;
-
-    for (let i = 0; i < mixins.length; i++) {
-      // Would just pass mixin, but want to leave those endpoints more open for now.
-      console.log(`registering Mixin: ${mixins[i].name}`);
-      registerMixin(mixins[i].mixin, mixins[i].name);
-    }
-  }
-
-  if (plugin.modules) {
-    const modules = plugin.modules;
-
-    for (let i = 0; i < modules.length; i++) {
-      // Would just pass modules, but want to leave those endpoints more open for now.
-      console.log(`registering Module: ${modules[i].name}`);
-      registerModule(modules[i].module, modules[i].name);
-    }
+  if (plugin.module) {
+    registerModule(plugin.name, plugin.module);
   }
 
-  if (plugin.tools) {
-    const tools = plugin.tools;
-
-    for (let i = 0; i < tools.length; i++) {
-      // Would just pass modules, but want to leave those endpoints more open for now.
-      registerTool(tools[i].tool, tools[i].name);
-    }
+  if (plugin.items) {
+    registerItems(plugin.name, plugin.items, overwrite)
   }
 
-  // TODO -> This is easily exensible.
-  // We could have e.g. 3.X -> "Plugins can now include manipulators, just add
-  // a manipulators array to your plugin file."
-  //
-  // TODO: Could also do version checking to see if the cTools version is recent
-  // enough to be compatible with the plugin.
-
-  pluginList.push(plugin.name);
 }
 
-
-function isPluginNameRegistered (name) {
-  return pluginList.some((registeredPluginName) => {
-    return registeredPluginName === name;
-  });
+function isPluginNameRegistered (namespace) {
+  return lib[namespace] !== undefined;
 }
```

Comment:
I have mixed feelings about this abstraction. I understand why and the convenience, I also think it will be much more brittle than only exposing import/register/registerModule

If you really like this, I can compromise, but I'd like to haggle with you on the schema for this